### ISSUE TITLE
No administrator removal mechanism leads to intermittent request failure 

### ISSUE THREAD

**Issue by anglinb** [education/classroom#1167]
**tl;dr** 👋 If an owner of an organization using classroom leaves the organization, they remain able to create assignments. Additionally, use of their auth token (now w/o org admin permissions) leads to intermittent failure for assignment creation. 

---

### Description
I believe that users who are owners of an organization used by classroom will never get removed from administrating the classroom even if they stop being owners on the organization. One could argue that this is not a big issue, as once you're an organization owner I don't think you can get "kicked" without voluntary self-removal. 

However, even if this functionality is working as expected, the `github_client` on the `Organization` model will not  have the correct permissions to (for example) create repositories when a user accepts an assignment: [`token = users.limit(1).order("RANDOM()").pluck(:token)[0]`](http://github.com/education/classroom/blob/94e678a029291f8d2c1775a9654f050e13864d04/app/models/organization.rb#L39-L40)

### Possible Solution

I believe an [Organization Webhook](https://developer.github.com/v3/orgs/hooks/) listening for [Membership Events](https://developer.github.com/v3/activity/events/types/#membershipevent) could allow Classroom to remove a user when they are removed from the organization. However, since webhooks are not always delivered, catching an authentication error for a given `organization`.`github_client` and verifying organization membership could keep the request failures minimal and ensure that user gets removed from administering the classroom. 

/cc @gregose

**Comment by anglinb** [comment]
:wave: friendly bump
